web/src/main/java/no/nav/modiapersonoversikt/consumer/dkif/DigDirServiceImpl.kt

```diff
@@ -0,0 +1,99 @@
+import okhttp3.OkHttpClient
+
+class DigDirServiceImpl(
+    baseUrl: String = EnvironmentUtils.getRequiredProperty("DIG_DIR_REST_URL"),
+    machineToMachineTokenClient: MachineToMachineTokenClient,
+    private val cache: Cache<Fnr, Dkif.DigitalKontaktinformasjon> = CacheUtils.createCache()
+) : Dkif.Service, Pingable {
+
+    private val scope = DownstreamApi.parse(EnvironmentUtils.getRequiredProperty("DIG_DIR_SCOPE"))
+
+    private val httpClient: OkHttpClient = RestClient.baseClient().newBuilder()
+        .addInterceptor(
+            LoggingInterceptor("digdir-krr-proxy") { request ->
+                requireNotNull(request.header("Nav-Call-Id"))
+            }
+        )
+        .addInterceptor(
+            AuthorizationInterceptor {
+                machineToMachineTokenClient.createMachineToMachineToken(scope)
+            }
+        )
+        .build()
+
+    private val client = PersonControllerApi(basePath = baseUrl, httpClient = httpClient)
+    private val pingApi = PingApi(baseUrl, httpClient)
+
+    override fun hentDigitalKontaktinformasjon(fnr: String): Dkif.DigitalKontaktinformasjon {
+        return requireNotNull(
+            cache.get(Fnr(fnr)) {
+                val kontaktInfoTilBruker = KontaktInformasjonResult.runCatching {
+                    client.getPerson(
+                        navPersonident = fnr,
+                        navCallId = getCallId(),
+                        inkluderSikkerDigitalPost = true
+                    )
+                }.getOrNull()
+
+                if (kontaktInfoTilBruker == null) {
+                    TjenestekallLogger.warn(
+                        header = "Feil ved henting av digital kontaktinformasjon fra digdir",
+                        fields = mapOf(
+                            "fnr" to fnr,
+                        )
+                    )
+                    Dkif.INGEN_KONTAKTINFO
+                } else {
+                    Dkif.DigitalKontaktinformasjon(
+                        personident = kontaktInfoTilBruker.personident,
+                        reservasjon = kontaktInfoTilBruker.reservert.toString(),
+                        epostadresse = Dkif.Epostadresse(
+                            value = kontaktInfoTilBruker.epostadresse,
+                            sistOppdatert = kontaktInfoTilBruker.epostadresseOppdatert?.toLocalDate(),
+                            sistVerifisert = kontaktInfoTilBruker.epostadresseVerifisert?.toLocalDate()
+                        ),
+                        mobiltelefonnummer = Dkif.MobilTelefon(
+                            value = kontaktInfoTilBruker.mobiltelefonnummer,
+                            sistOppdatert = kontaktInfoTilBruker.mobiltelefonnummerOppdatert?.toLocalDate(),
+                            sistVerifisert = kontaktInfoTilBruker.mobiltelefonnummerVerifisert?.toLocalDate()
+                        )
+                    )
+                }
```

Comment:
Tror ikke vi trenger å introdusere en ny `Result` type for dette. 

Tipper man kan gjøre noe alà;
```kotlin
cache.get(Fnr(fnr)) {
  client.runCatching {
    getPerson(
      navPersonident = fnr, 
      navCallId = getCallId(), 
      inkluderSikkerDigitalPost = true
    )
  }.map{ data ->
    Dkif.DigitalKontaktinformasjon(
      personident = data.personident
      epostadresse = TODO()
      // OSV
    )
  }.getOrElse { exception ->
    TjenestekallLogger.warn(
      header = "Feil ved henting av digital kontaktinformasjon fra digdir"),
      fields = mapOf(
        "exception" to exception
        "fnr" to fnr
      )
    )
    Dkif.INGEN_KONTAKTINFO
  }
}
```

Reply:
enig